app/src/androidTest/java/pl/makimo/androidtemplate/ExampleInstrumentedTest.kt

```diff
@@ -3,24 +3,34 @@ package pl.makimo.androidtemplate
 import androidx.test.rule.ActivityTestRule
 
 import org.junit.Test
+import org.junit.Rule
+import org.junit.Before
 import org.junit.runner.RunWith
-
 import org.junit.Assert.*
-import org.junit.Rule
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import androidx.test.platform.app.InstrumentationRegistry
+import pl.makimo.androidtemplate.retrofit.ApiService
 
 /**
- * Instrumented test, which will execute on an Android device.
- *
- * See [testing documentation](http://d.android.com/tools/testing).
+ * Boilerplate test for checking build configuration.
  */
 @RunWith(AndroidJUnit4::class)
-class ExampleInstrumentedTest {
+class BuildConfigTest {
     @Rule
     @JvmField
     val rule: ActivityTestRule<MainActivity> = ActivityTestRule(MainActivity::class.java)
 
+    @Before
+    fun apiServiceBuilds() {
+        try {
+            ApiService.get()
+        } catch (e: Throwable) {
+            assert(false)
+        } finally {
+            assert(true)
```

Comment:
This one is not so obvious. You call assert with false in catch block and after that you call assert with true in finally block? As far as I understood this code it would be more readable to put assert with true in try block.

Reply:
:roll_eyes: